src/scale.c

```diff
@@ -180,3 +182,33 @@ avifBool avifImageScale(avifImage * image,
 }
 
 #endif
+
+avifResult avifImageCopySamples(avifImage * dstImage, const avifImage * srcImage, avifPlanesFlags planes)
+{
+    for (int c = AVIF_CHAN_Y; c <= AVIF_CHAN_A; ++c) {
+        if ((!(planes & AVIF_PLANES_YUV) && (c != AVIF_CHAN_A)) || (!(planes & AVIF_PLANES_A) && (c == AVIF_CHAN_A))) {
+            continue;
+        }
```

Comment:
Nit/optional: This if statement (lines 189-191) is a little hard to understand. I seem to remember you used an alternative solution like the following:
```
    int firstPlane = AVIF_CHAN_A;
    int lastPlane = AVIF_CHAN_V;
    if (planes & AVIF_PLANES_YUV) {
        firstPlane = AVIF_CHAN_Y;
    }
    if (planes & AVIF_PLANES_A) {
        lastPlane = AVIF_CHAN_A;
    }
    for (int c = firstPlane; c <= lastPlane; ++c) {
```
But I can't find it in the source tree. Maybe it was deleted or rewritten during the code review of the pull request.

The alternative solution is also a little hard to understand, but it is easier to parse.